### PR TITLE
virtio_fs: restruct virtiofs memory backend test

### PR DESCRIPTION
**Pull request by xiagao**
ID: 2193087
1. add memfd backend.
2. test hugepage on all backend.
3. test numa on all backend.

### CODE REVIEW

**Comment by xiagao** [comment]
(1/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.q35: STARTED
 (1/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.q35: FAIL: Could not get virtio-fs mounted volume letter. (263.40 s)
 (2/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_numa.q35: STARTED
 (2/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_numa.q35: PASS (234.80 s)
 (3/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.q35: STARTED
 (3/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.q35: PASS (150.14 s)
 (4/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.with_numa.q35: STARTED
 (4/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.with_numa.q35: PASS (150.95 s)
 (5/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.q35: STARTED
 (5/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.q35: FAIL: Could not get virtio-fs mounted volume letter. (263.18 s)
 (6/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_numa.q35: STARTED
 (6/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_numa.q35: PASS (176.38 s)
 (7/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.q35: STARTED
 (7/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.q35: PASS (152.37 s)
 (8/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.with_numa.q35: STARTED
 (8/8) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.with_numa.q35: PASS (179.80 s)

**Comment by xiagao** [comment]
@zhenyzha Could you help  to check if multi-arch need to test those scenarios?

**Comment by zhenyzha** [comment]
doesn't look as expected:

git am or without
https://github.com/avocado-framework/avocado-vt/pull/3666/files

The result is failure

```
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.arm64-pci: ERROR
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.arm64-pci: ERROR
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.with_numa.arm64-pci: ERROR
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.arm64-pci: ERROR
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.arm64-pci: ERROR
Host_RHEL.m9.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.with_numa.arm64-pci: ERROR
```

The specific log is uploaded to the bug

**Comment by xiagao** [comment]
Hi @zhenyzha I updated this patch, could you have a check? Currently, there is numa and hugepage supported.

 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_numa.q35: STARTED
 (1/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_numa.q35: PASS (85.74 s)
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.with_numa.q35: STARTED
 (2/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.file_backend.with_hugepage.with_numa.q35: PASS (88.21 s)
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_numa.q35: STARTED
 (3/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_numa.q35: PASS (89.42 s)
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.with_numa.q35: STARTED
 (4/4) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.memfd_backend.with_hugepage.with_numa.q35: PASS (87.37 s)

**Comment by leidwang** [comment]
Hi @zhenyzha Could you please help to review this patch when you are free?Thanks!

**Comment by zhenyzha** [comment]
After discussing in the automation meeting, 
Maintainer feels that it is safer to wait for the vt [patch](https://github.com/avocado-framework/avocado-vt/pull/3666/files)  to be merged before verifying the patch

File: qemu/tests/cfg/virtio_fs_memory_backend.cfg [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            start_vm = no
+            setup_hugepages = yes
+            pre_command_noncritical = yes
+            pre_command = 'sync && echo 3 > /proc/sys/vm/drop_caches && echo 1 > /proc/sys/vm/compact_memory'
+    variants:
+        - file_backend:
+            backend_mem_mem0 = memory-backend-file
+            variants:
+                - @default:
+                - with_hugepage:
```
Comment (zhenyzha) [inline qemu/tests/cfg/virtio_fs_memory_backend.cfg:89]: This is repeated in line 79

**Comment by xiagao** [comment]
(1/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.q35: STARTED
 (1/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.q35: PASS (119.86 s)
 (2/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.q35: STARTED
 (2/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.q35: PASS (119.12 s)
 (3/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.q35: STARTED
 (3/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.q35: PASS (222.85 s)
 (4/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.q35: STARTED
 (4/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.q35: PASS (188.36 s)
 (5/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.q35: STARTED
 (5/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.q35: PASS (130.96 s)
 (6/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.q35: STARTED
 (6/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.q35: PASS (125.90 s)

File: qemu/tests/cfg/virtio_fs_memory_backend.cfg (hunk 1 above)
Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_memory_backend.cfg:87]: update， thanks

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

git am 3709.patch
Applying: virtio_fs: restruct virtio fs memory backend test

```
 (1/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.arm64-pci: STARTED
 (1/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.arm64-pci: PASS (133.33 s)
 (2/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.arm64-pci: STARTED
 (2/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.arm64-pci: PASS (125.83 s)
 (3/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.arm64-pci: STARTED
 (3/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.arm64-pci: PASS (130.28 s)
 (4/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.arm64-pci: STARTED
 (4/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.arm64-pci: PASS (132.68 s)
 (5/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.arm64-pci: STARTED
 (5/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.arm64-pci: PASS (129.31 s)
 (6/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.arm64-pci: STARTED
 (6/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.arm64-pci: PASS (127.71 s)
```

**Comment by xiagao** [comment]
(1/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.q35: STARTED
 (1/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.q35: PASS (169.90 s)
 (2/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.q35: STARTED
 (2/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.q35: PASS (122.73 s)
 (3/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.q35: STARTED
 (3/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.q35: PASS (212.17 s)
 (4/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.q35: STARTED
 (4/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.q35: PASS (183.15 s)
 (5/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.q35: STARTED
 (5/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.q35: PASS (134.77 s)
 (6/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.q35: STARTED
 (6/6) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.q35: PASS (123.93 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

@fbq815 Could you help check this patch? Thx.

**Comment by fbq815** [comment]
s390x:
JOB ID     : 09fcea354344927eb477cb3a75d70818f93e3e20
JOB LOG    : /root/avocado/job-results/job-2023-09-17T22.24-09fcea3/job.log
 (1/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (425.77 s)
 (2/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: STARTED
 (2/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: PASS (45.35 s)
 (3/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.s390-virtio: STARTED
 (3/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.s390-virtio: ERROR: VM creation command failed:    'MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm -S  -name \'avocado-vt-vm1\'  -sandbox on -machine s390-ccw-virtio  -nodefaults  -vga none -m 11264 -object \'{"size": 11811160064, "mem-path": "/dev/shm", "share": true, "id": "mem-... (7.44 s)
 (4/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: STARTED
 (4/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: ERROR: local variable 'os_type' referenced before assignment (9.28 s)
 (5/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.s390-virtio: STARTED
 (5/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.s390-virtio: ERROR: local variable 'os_type' referenced before assignment (8.56 s)
 (6/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: STARTED
 (6/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: PASS (43.85 s)
 (7/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.s390-virtio: STARTED
 (7/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.s390-virtio: ERROR: VM creation command failed:    'MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm -S  -name \'avocado-vt-vm1\'  -sandbox on -machine s390-ccw-virtio  -nodefaults  -vga none -m 11264 -object \'{"size": 11811160064, "share": true, "id": "mem-mem0", "qom-type": "memo... (7.39 s)
RESULTS    : PASS 3 | ERROR 4 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-09-17T22.24-09fcea3/results.html
JOB TIME   : 552.89 s

@xiagao please help me remove or substitute those four unsupported testcases on s390x
we don't support NUMA and 1M memory-backend

**Comment by xiagao** [comment]
> s390x: JOB ID : 09fcea354344927eb477cb3a75d70818f93e3e20 JOB LOG : /root/avocado/job-results/job-2023-09-17T22.24-09fcea3/job.log (1/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED (1/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (425.77 s) (2/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: STARTED (2/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: PASS (45.35 s) (3/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.s390-virtio: STARTED (3/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_numa.s390-virtio: ERROR: VM creation command failed: 'MALLOC_PERTURB_=1 /usr/libexec/qemu-kvm -S -name 'avocado-vt-vm1' -sandbox on -machine s390-ccw-virtio -nodefaults -vga none -m 11264 -object '{"size": 11811160064, "mem-path": "/dev/shm", "share": true, "id": "mem-... (7.44 s) (4/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: STARTED (4/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: ERROR: local variable 'os_type' referenced before assignment (9.28 s) (5/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.s390-virtio: STARTED (5/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.with_numa.s390-virtio: ERROR: local variable 'os_type' referenced before assignment (8.56 s) (6/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: STARTED (6/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: PASS (43.85 s) (7/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.s390-virtio: STARTED (7/7) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.with_numa.s390-virtio: ERROR: VM creation command failed: 'MALLOC_PERTURB_=1 /usr/libexec/qemu-kvm -S -name 'avocado-vt-vm1' -sandbox on -machine s390-ccw-virtio -nodefaults -vga none -m 11264 -object '{"size": 11811160064, "share": true, "id": "mem-mem0", "qom-type": "memo... (7.39 s) RESULTS : PASS 3 | ERROR 4 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0 JOB HTML : /root/avocado/job-results/job-2023-09-17T22.24-09fcea3/results.html JOB TIME : 552.89 s
> 
> @xiagao please help me remove or substitute those four unsupported testcases on s390x we don't support NUMA and 1M memory-backend

Update, please have a check.

**Comment by fbq815** [comment]
JOB ID     : 4b63b9cea64562497b69cba33b0b49cd6a33644b
JOB LOG    : /root/avocado/job-results/job-2023-09-19T03.28-4b63b9c/job.log
 (1/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: STARTED
 (1/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.s390-virtio: PASS (44.89 s)
 (2/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: STARTED
 (2/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_file.with_hugepage.s390-virtio: PASS (44.94 s)
 (3/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: STARTED
 (3/3) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.virtio_fs_memory_backend.backend_memfd.s390-virtio: PASS (43.72 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-09-19T03.28-4b63b9c/results.html
JOB TIME   : 135.86 s

LGTM

**Review by fbq815 — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]

**Comment by YongxueHong** [comment]
LGTM.

**Pull request merged**